### PR TITLE
Pngquantlib

### PR DESCRIPTION
**Pull request by kodabb**
This PR updates the pngquant in order to install development files and offer different configure options.

I am not familiar with homebrew releases, but I believe a new bottle should be made after the first commit.

### CODE REVIEW

**Comment by mistydemeo** [comment]
Is it worth bumping the revision to ensure all users get this, or are most users unlikely to need libimagequant?

**Comment by kodabb** [comment]
It's just for development, so most likely users who need those files already compiled their own version ;)
So for me it's fine to wait for next bump.

File: Library/Formula/pngquant.rb [hunk 1]
```diff
@@ -14,12 +14,35 @@ class Pngquant < Formula
 
   depends_on "pkg-config" => :build
   depends_on "libpng"
+  depends_on "little-cms2" => :optional
+
+  option "with-openmp", "Enable OpenMP"
+  if build.with? "openmp"
```
Comment (mistydemeo) [inline Library/Formula/pngquant.rb]: We have sugar for this: `needs :openmp`

This could be better represented as `needs :openmp if build.with? "openmp"`

Comment (mistydemeo) [inline Library/Formula/pngquant.rb]: (Sorry the docs are sparse on this point!)

**Comment by mistydemeo** [comment]
Do you think it's better for users of this software to offer an explicit `with-openmp` option, or silently build with OpenMP wherever it's available without asking?

**Comment by kodabb** [comment]
I think it would be better to have openmp enabled, however I am not sure how to express that since it only works when gcc is used. Will the 'fails_with' interact well with 'needs'?

**Comment by mistydemeo** [comment]
Yes, the `needs` is sugar which constructs the appropriate `fails_with` statements behind the scenes. `needs :openmp` actually expands to a `fails_with :clang` statement.

**Comment by kodabb** [comment]
ok so i tried with
```
  option "without-openmp", "Disable OpenMP support"
  needs :openmp if build.with? "openmp" # only builds with GCC
```
however without gcc installed it will yield to 
```
Error: pngquant cannot be built with any available compilers.
To install this formula, you may need to:
  brew install gcc
```
which might be overkill.
Is there a way to notify the user that he could still build with `--without-openmp`, or just disable it automatically if gcc is not found?
Otherwise I'm thinking it's better to keep it disabled. What do you suggest?

**Comment by mistydemeo** [comment]
That works; you can remove the class-level stuff and make the conditional `if !ENV.compiler == :clang` instead. If you think it's important enough to notify the user it's building without OpenMP, you could add conditional caveats to note that it was built without OpenMP support, maybe?

**Comment by kodabb** [comment]
Sorry I wrote too many options there, which one "works"? :D

**Comment by kodabb** [comment]
Ok, I thought about it and figured out that the current version with an optional openmp dependency and the `:needs` clause is the best trade off in usability and maintenance. Patch ready to go from my side.

**Comment by kodabb** [comment]
ping?

**Comment by mikemcquaid** [comment]
Please use `needs :openmp if build.with? "openmp"` as @mistydemeo suggested, thanks!

**Comment by bfontaine** [comment]
Any update on this?

**Comment by kodabb** [comment]
kinda lost track of this, i should be able to look at it next week hopefully

**Comment by DomT4** [comment]
Any update?

**Comment by kodabb** [comment]
thanks for the reminder, situation slightly improved in 2.5, now it should be so straightforward that I enabled openmp by default

**Comment by kodabb** [comment]
hmm i forgot about static linking and how openmp by default by be a headache for users of the library only, so i switched it back to off by default again

**Comment by kodabb** [comment]
any more changes needed before this can go in?

File: Library/Formula/pngquant.rb [hunk 2]
```diff
@@ -14,12 +14,28 @@ class Pngquant < Formula
...
+      args << "--with-openmp"
+      args << "--without-cocoa"
+    end
+
+    system "./configure", *args
+    system "make", "install", "CC=#{ENV.cc}"
+
     man1.install "pngquant.1"
+    lib.install "lib/libimagequant.a"
+    include.install "lib/libimagequant.h"
```
Comment (bfontaine) [inline Library/Formula/pngquant.rb]: What does `make install` do if we need to install these by ourselves?

Comment (kodabb) [inline Library/Formula/pngquant.rb]: it will install the binary only

Comment (mikemcquaid) [inline Library/Formula/pngquant.rb]: I guess we wonder if this is a private header?

Comment (kodabb) [inline Library/Formula/pngquant.rb]: it's not...
https://pngquant.org/lib/

Comment (mikemcquaid) [inline Library/Formula/pngquant.rb]: :+1:

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**